include/zephyr/pm/state.h

```diff
@@ -182,6 +182,22 @@ struct pm_state_info {
 	COND_CODE_1(DT_NODE_HAS_STATUS(DT_PHANDLE_BY_IDX(node_id, cpu_power_states, i), okay),    \
 		    (PM_STATE_INFO_DT_INIT(DT_PHANDLE_BY_IDX(node_id, cpu_power_states, i)),), ())
 
+/**
+ * @brief Helper macro to initialize an entry of a struct pm_state_info array
+ * when using UTIL_LISTIFY in PM_STATE_INFO_LIST_FROM_DT_REINIT.
+ *
+ * @note Only enabled states are initialized.
+ *
+ * @param i UTIL_LISTIFY entry index.
+ * @param node_id A node identifier with compatible zephyr,power-state
+ */
+#define Z_PM_STATE_INFO_FROM_DT_REINIT(i, node_id)						  \
+	COND_CODE_1(DT_NODE_HAS_STATUS(DT_PHANDLE_BY_IDX(node_id, st_reinit_power_states, i),	  \
+				       okay),							  \
+		    (PM_STATE_INFO_DT_INIT(DT_PHANDLE_BY_IDX(node_id, st_reinit_power_states,	  \
+							     i)),),				  \
+		    ())
```

Comment:
this should definitely be part of an internal ST header (something similar existed for pinmux before pinctrl existed, maybe you can check old releases like 2.6)

Reply:
Moved the macros to `include/zephyr/devicetree/pm_stm32.h`. I'd decided on `pm_stm32.h` as it may later be expanded with general STM32 pm exception cases, not just state-specific ones.